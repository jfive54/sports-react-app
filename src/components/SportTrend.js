import React from "react";

import { VtmnTextInput } from "@vtmn/react";

import configData from "../config.json";
import Sports from "./Sports";

class SportList extends React.Component {
  constructor(props) {
    super(props);
    this.state = { items: [], filterItems: [], text: "" };
    this.handleChange = this.handleChange.bind(this);
  }

  render() {
    return (
      <div>
        <div className="vtmn-flex vtmn-bg-brand-digital-light-3 vtmn-rounded-lg vtmn-p-6 vtmn-mb-5">
          <p className="vtmn-text-xl vtmn-text-center vtmn-font-semibold vtmn-text-black">
            Sport
            <span className="vtmn-text-brand-digital vtmn-text-2xl">
              Trend <span className="vtmx-temp-hot-line inline-icon"></span>
            </span>
          </p>
        </div>
        <div className="vtmn-flex vtmn-justify-around">
          <div>
            <VtmnTextInput
              labelText="Filtrer"
              onChange={this.handleChange}
              identifier="filter"
              helperText="Filtrer les sports"
            />
          </div>
        </div>
        <div className="vtmn-flex vtmn-justify-around">
          <Sports items={this.state.filterItems} />
        </div>
      </div>
    );
  }

  componentDidMount() {
    fetch(configData.SPORT_LIST_URL)
      .then((res) => {
        return res.json();
      })
      .then(({ data }) => {
        this.setState({ items: data, filterItems: this.sortData(data) });
      })
      .catch(console.log);
  }

  sortData(data) {
    return data.sort((a, b) =>
      a.attributes?.name > b.attributes?.name
        ? 1
        : b > a.attributes?.name
        ? -1
        : 0
    );
  }

  cleanSearch(search) {
    return search
      .normalize("NFD")
      .replace(/[\u0300-\u036f]/g, "")
      .toUpperCase();
  }

  handleChange(e) {
    let itemsCopy = [...this.state.items];
    const filterItems = itemsCopy.filter((sport) =>
      this.cleanSearch(sport.attributes?.name ?? "").includes(
        this.cleanSearch(e.target.value)
      )
    );

    this.setState({
      text: e.target.value,
      filterItems: this.sortData(filterItems),
    });
  }
}

export default SportList;
