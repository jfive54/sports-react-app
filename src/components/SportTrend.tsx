import React from "react";

import { VtmnTextInput } from "@vtmn/react";

import { getSports } from "../services/sports-service";

import Sports from "./Sports";
import { Item } from "./interfaces";

interface Props {}
interface State {
  items: Item[];
  filterItems: Item[];
  text: string;
}

class SportList extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { items: [], filterItems: [], text: "" };
    this.handleChange = this.handleChange.bind(this);
  }

  async componentDidMount() {
    const { data } = await getSports();
    this.setState({ items: data, filterItems: this.sortData(data) });
  }

  sortData(data: Item[]) {
    return data.sort((a: Item, b: Item) =>
      a.attributes?.name > b.attributes?.name
        ? 1
        : b.attributes?.name > a.attributes?.name
        ? -1
        : 0
    );
  }

  cleanSearch(search: string) {
    return search
      .normalize("NFD")
      .replace(/[\u0300-\u036f]/g, "")
      .toUpperCase();
  }

  handleChange(e: any) {
    let itemsCopy = [...this.state.items];
    const filterItems = itemsCopy.filter((sport) =>
      this.cleanSearch(sport.attributes?.name ?? "").includes(
        this.cleanSearch(e.target.value)
      )
    );

    this.setState({
      text: e.target.value,
      filterItems: this.sortData(filterItems),
    });
  }

  render() {
    return (
      <div>
        <div className="vtmn-flex vtmn-bg-brand-digital-light-3 vtmn-rounded-lg vtmn-p-6 vtmn-mb-5">
          <p className="vtmn-text-xl vtmn-text-center vtmn-font-semibold vtmn-text-black">
            Sport
            <span className="vtmn-text-brand-digital vtmn-text-2xl">
              Trend <span className="vtmx-temp-hot-line inline-icon"></span>
            </span>
          </p>
        </div>
        <div className="vtmn-flex vtmn-justify-around">
          <div>
            <VtmnTextInput
              labelText="Filtrer"
              onChange={this.handleChange}
              identifier="filter"
              helperText="Filtrer les sports"
            />
          </div>
        </div>
        <div className="vtmn-flex vtmn-justify-around">
          <Sports items={this.state.filterItems} />
        </div>
      </div>
    );
  }
}

export default SportList;
