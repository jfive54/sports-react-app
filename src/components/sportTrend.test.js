import React from "react";
import renderer from "react-test-renderer";

import SportTrend from "./SportTrend";

describe("Sport trend", () => {
  test("Sport trend test", async () => {
    const component = await renderer.create(<SportTrend />);

    let sportTrend = component.toJSON();
    expect(sportTrend).toMatchSnapshot();
  });
});
