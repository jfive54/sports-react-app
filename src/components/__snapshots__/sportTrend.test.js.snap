// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sport trend Sport trend test 1`] = `
<div>
  <div
    className="vtmn-flex vtmn-bg-brand-digital-light-3 vtmn-rounded-lg vtmn-p-6 vtmn-mb-5"
  >
    <p
      className="vtmn-text-xl vtmn-text-center vtmn-font-semibold vtmn-text-black"
    >
      Sport
      <span
        className="vtmn-text-brand-digital vtmn-text-2xl"
      >
        Trend 
        <span
          className="vtmx-temp-hot-line inline-icon"
        />
      </span>
    </p>
  </div>
  <div
    className="vtmn-flex vtmn-justify-around"
  >
    <div>
      <label
        className="vtmn-text-input_label"
        htmlFor="filter"
      >
        Filtrer
      </label>
      <div
        className="vtmn-text-input_container"
      >
        <input
          className="vtmn-text-input"
          disabled={false}
          id="filter"
          onChange={[Function]}
          type="text"
        />
      </div>
      <p
        className="vtmn-text-input_helper-text"
      >
        Filtrer les sports
      </p>
    </div>
  </div>
  <div
    className="vtmn-flex vtmn-justify-around"
  >
    <ul />
  </div>
</div>
`;
